frontend/src/modal/ModalTemplate.jsx

```diff
@@ -0,0 +1,43 @@
+import React, { useEffect, useState } from 'react';
+
+import './modal.css';
+
+function Modal(props) {
+  const [opacity, setOpacity] = useState(0);
+
+  useEffect(() => {
+    setOpacity(1);
+  }, []);
+
+  function fadeOut() {
+    setOpacity(0);
+  };
+
+  return (
+    <div className='modal' hidden={props.hidden} onClick={props.close}>
```

Comment:
I would recommend using a `<dialog>` element instead. It gives a lot of functionality out of the box.

[Modals made easy](https://www.youtube.com/watch?v=TAB_v6yBXIE&t=269s&ab_channel=KevinPowell)
[Styling modal](https://www.youtube.com/watch?v=RO40Opyizeg&ab_channel=KevinPowell)